### DISCUSSION TITLE
How to use YAML for a list of redirects

### DISCUSSION THREAD

**Discussion by Vacilando** [gatsbyjs/gatsby#34074] [Help] [answered]
At https://support.gatsbyjs.com/hc/en-us/articles/1500003051241-Working-with-Redirects it says 

> we use a YAML file to store our redirects

but the example given uses JSON.

What is the syntax and the file name in case we wish to use a YAML file?
Could it also be added to that documentation page?

**Comment by LekoArts** [comment, marked as answer]
You can use a `const redirects = loadYaml('./path/to.yaml')` with e.g.

```js
const yaml = require(`js-yaml`)
const fs = require(`fs`)

/**
 * Import yaml in server-side code in a way that works when the site
 * is run as a theme.
 *
 * @example
 * ```
 * const docLinks = loadYaml(`src/data/sidebars/doc-links.yaml`)
 * ```
 */
function loadYaml(yamlPath) {
  return yaml.safeLoad(fs.readFileSync(yamlPath, `utf8`))
}

module.exports = { loadYaml }
```

**Comment by Vacilando** [comment]
Nice, @LekoArts!
Could you also give an example of how the redirects should be specified in the YAML file?

**Reply by LekoArts** [reply]
Something like this:

```yaml
- fromPath: /foo/
  toPath: /bar/
- fromPath: /abc/
  toPath: /def/
```

**Comment by Vacilando** [comment]
FYI:

> Error: Function yaml.safeLoad is removed in js-yaml 4. Use yaml.load instead, which is now safe by default.
